### PR TITLE
Fixes to build scripts for initial release

### PR DESCRIPTION
**Pull request by omus**
In attempting to do our initial release I encountered a few issues with things being out of date.

As I worked through these issues I noticed that the artifacts we are uploading have very specific triplets. Normally, the triplets from BinaryBuilder are based upon `BinaryBuilder.supported_platforms()` which uses quite generic triplets and mainly focuses on OS and architecture. Having an overly specific triplet is an issue in that it can result in us generating more artifacts than we have to. I've reduced the triplet down to what should be the current minimum. Ideally, we wouldn't use the Julia version in our triplet but appears that our `WORKSPACE.bazel` setup is pulling in Julia specific headers which is resulting in shared libraries which are incompatible with other Julia version (I think it's because our library is fully statically compiled).

Anyway, these should be the final changes needed to be merged so we can cut our initial release.

### CODE REVIEW

File: build/bind_artifacts.jl
Comment (omus) [inline build/bind_artifacts.jl:30]: By using a list of required platforms we can ensure that if we failed to upload all of the artifacts that the `bind_artifacts.jl` script will fail as it will attempt to download a file that doesn't exist

File: build/bind_artifacts.jl
Comment (omus) [inline build/bind_artifacts.jl:36]: Previously we downloaded files to a temporary directory instead of the `TARBALL_DIR`. This will override any locally generated files but that seems fine.

File: build/common.jl
Comment (omus) [inline build/common.jl:28]: Didn't spend much time refactoring this. If we can make the artifacts non-Julia specific this becomes much cleaner

File: build/common.jl
Comment (omus) [inline build/common.jl:46]: @glennmoy I can understand your reason for copying the regex from LibGit2 as it's not exported by the library. However, now that I'm using `git_url` as well it seemed better to use the function and regex from the library that tests them.

File: build/common.jl
Comment (omus) [inline build/common.jl:52]: By using `Base.thisminor` we ensure that the `julia_version` is always `X.Y.0` which produces a more consistent artifact URL

File: build/upload_tarballs.jl
Comment (omus) [inline build/upload_tarballs.jl:105]: The `try`/`catch` here not only suppressed `ghr` errors but also the `throw`n errors from the Julia code

**Comment by omus** [comment]
@glennmoy feel free to approve and merge this when you get in. Once that's done we can make the `v0.0.1` release.

**Comment by glennmoy** [comment]
>As I worked through these issues I noticed that the artifacts we are uploading have very specific triplets. Normally, the triplets from BinaryBuilder are based upon BinaryBuilder.supported_platforms() which uses quite generic triplets and mainly focuses on OS and architecture. Having an overly specific triplet is an issue in that it can result in us generating more artifacts than we have to.

Yeah I wasn't sure about having to include `libgfortran` etc since we're doing this outside BinaryBuilder...I went looking for alternative functions but concluded that perhaps `triplet()` was the canonical way to generate the artifact names? Seems it can be overly specific but maybe that's by design so the user can decide what not to include?

File: build/bind_artifacts.jl
Comment (glennmoy) [inline build/bind_artifacts.jl:37]: will this fail if it tries to fetch an artifact that hasn't been uploaded yet?

File: build/common.jl
Comment (glennmoy) [reply build/common.jl:28]: LGTM - it's not worth prettifying anyway

File: build/common.jl
Comment (glennmoy) [reply build/common.jl:46]: So my understandig is this converts, e.g. `git@github.com:beacon-biosignals/ray.jl` to `https://github.com:beacon-biosignals/ray.jl`?

File: build/common.jl
Comment (glennmoy) [inline build/common.jl:53]: odd that the keys get converted to strings but the must be provided as kwargs...

File: build/common.jl
Comment (glennmoy) [inline build/common.jl:62]:
```suggestion
    return "ray_julia.$tag.$(triplet(supported_platform(platform))).tar.gz"
```

I wonder do we want to call `supported_platform` again here for extra protection...although this function is only called over the `SUPPORTED_PLATFORMS` anyway it's probably overkill

File: build/upload_tarballs.jl
Comment (glennmoy) [inline build/upload_tarballs.jl:65]: first time I wrote this I wanted to see if there was a `file` command (similar to the bash command) that would assert the file type rather than check against a regex. I went looking in `Base.Filesystem` module but didn't see anything. Only calling it out in case you knew of one off hand - not suggesting we have to change it now.

**Review by glennmoy — APPROVED** [review]
I'll give this a run through to make sure it works before merging

**Comment by glennmoy** [comment]
Assuming this works we'll merge it, I'll regenerate the Artifacts and then we can make a tag!

File: build/bind_artifacts.jl
Comment (glennmoy) [reply build/bind_artifacts.jl:37]: I checked and it does 👍

File: build/common.jl
Comment (omus) [reply build/common.jl:46]: Yes, it converts SCP-like URLs to HTTPS URLs. Specifically `git@github.com:beacon-biosignals/Ray.jl` to `https://github.com/beacon-biosignals/Ray.jl` (there was a syntax issue with your HTTPS URL).

**Comment by glennmoy** [comment]
@omus the build worked - I just have a few suggestions incoming

File: build/upload_tarballs.jl
Comment (omus) [reply build/upload_tarballs.jl:65]: I think the usual way of handling this would be just to look at the extension. The other way is probably to use FileIO.jl and use the detection function which would detect the file type from magic bytes in the header: https://juliaio.github.io/FileIO.jl/stable/registry/#Registry-table

File: build/upload_tarballs.jl
Comment (omus) [inline build/upload_tarballs.jl]: Using `all` here is a bit strange due to having an internal error. A `for` or this seems better:
```suggestion
    stale = filter(!startswith("ray_julia.$TAG"), readdir("tarballs"))
    isempty(stale) || error("Stale tarball(s) found:\n$(join(stale, '\n'))")
```

File: build/upload_tarballs.jl
Comment (omus) [reply build/upload_tarballs.jl:67]: Never mind. You're using the match groups later

**Pull request merged**
